src/Nuget.PackageIndex/NupkgLocalPackageLoader.cs

```diff
@@ -46,103 +46,103 @@ internal class NupkgLocalPackageLoader : ILocalPackageLoader
+                        //      So dlls under contract would have all package's target frameworks, dlls under fx folder would have 
+                        //      only that particular target fx.
+                        //
+                        // New nuget format:
+                        //      - assemblies live under lib\fx folders as before
+                        //      - instdead of contract folder there is now Ref folder which contains common contracts. Unlice old contract
+                        //        folder, Ref folder might contain subfolder Any meaning that those contracts belong to any target fx; and
+                        //        corresponding Fx subfolder, which would contain types specific to particular FX.
+                        // 
+                        //      Logic for new format:
+                        //       - if there is a ref folder use assemblies under ref\any and then ref\fx; ignore lib assemblies
+                        //       - if there no ref folder just use assemblies from lib\fx subfolders
+                        //  
+                        // Note: just in case we process all assemblies lib, ref and contract. Assemblies with ref/contract that are in 
+                        // lib folder would not contain any types normally, so they shoudl be quick to process.
+
+                        var packageTargetFrameworks = package.GetSupportedFrameworks().Select(x => VersionUtility.GetShortFrameworkName(x));
+                        var assembliesMetadata = new List<AssemblyMetadata>();
+                        foreach (var assembly in allAssemblies)
                         {
-                            continue;
-                        }
+                            var assemblyFullPath = Path.Combine(packageFolder, assembly.Path);
+                            if (!File.Exists(assemblyFullPath))
+                            {
+                                assemblyFullPath = Path.Combine(packageFolder, Path.GetFileNameWithoutExtension(nupkgFilePath), assembly.Path);
+                            }
 
-                        IEnumerable<string> assemblyTargetFrameworks =  assembly.SupportedFrameworks.Select(x => VersionUtility.GetShortFrameworkName(x));
-                        if (!assemblyTargetFrameworks.Any())
-                        {
-                            assemblyTargetFrameworks = packageTargetFrameworks;
-                        }
+                            if (!File.Exists(assemblyFullPath))
+                            {
+                                continue;
+                            }
 
-                        assembliesMetadata.Add(
-                            new AssemblyMetadata
+                            IEnumerable<string> assemblyTargetFrameworks =  assembly.SupportedFrameworks.Select(x => VersionUtility.GetShortFrameworkName(x));
```

Comment:
Aha yeah.. I just debugged it and indeed assemblies under ref don't have supported frameworks list... In this case i default to all frameworks supported by package. This would show all types for all fxs, even if type does not exist in portable in this example ... 
I used to have this pseudo code there:

if (supportedfx == null)
{
    var fxname = Path.GetFileName(Path.GetDirectoryName(assembly.Path));
    if (fxname == "tools")
    {
         continue;
    }
    if (fxname == "any")
   {
        supportedfx  = package supported fxs
   }
   else {
       supportedfxs = new [] { fxname };
  }
}

but it is very tied with the format and if something changes it could be broken, thats why i decided to use package supported frameworks list in case i don't have it for a particular assembly and live with potential rare cases when it might suggest type for assembly version for fx where it does not exist ... I checked the dump of the index having our default set of packages, there not so many such types.

So what you would suggest here? Hard depndency on a format or some deficiency in the logic?